tools/msg/staff/msgdisStaff.py

```diff
@@ -0,0 +1,231 @@
+            0x15: ("CMD_BACKGROUND", self.decode_cmd_background),
+            0x16: ("CMD_MARATHONTIME", self.decode_cmd_arg_none),
+            0x17: ("CMD_RACETIME", self.decode_cmd_arg_none),
+            0x18: ("CMD_POINTS", self.decode_cmd_arg_none),
+            0x1A: ("CMD_UNSKIPPABLE", self.decode_cmd_arg_none),
+            0x1B: ("CMD_TWO_CHOICE", self.decode_cmd_arg_none),
+            0x1C: ("CMD_THREE_CHOICE", self.decode_cmd_arg_none),
+            0x1D: ("CMD_FISH_INFO", self.decode_cmd_arg_none),
+            0x1E: ("CMD_HIGHSCORE", self.decode_cmd_highscore),
+            0x1F: ("CMD_TIME", self.decode_cmd_arg_none),
+        }
+        if self.decodedText != "":
+            return
+
+        prevText = False
+        prevNewline = True
+        prevCmd = False
+        self.decodePos = 0
+        textLen = len(self.text)
+        while self.decodePos < textLen:
+            char = self.text[self.decodePos]
+            self.decodePos += 1
+
+            if char >= 0x20 and char <= 0xAF: # Regular Characters
+                if prevCmd:
+                    self.decodedText += ' "'
+                elif prevNewline:
+                    self.decodedText += '"'
+
+                if char == 0x22: # Handle escaping "
+                    self.decodedText += '\\"'
+                else:
+                    self.decodedText += chr(char)
+
+                prevText = True
+                prevNewline = False
+                prevCmd = False
+            elif char == 0x1: # New line
+                if prevCmd:
+                    self.decodedText += ' "'
+                elif prevNewline:
+                    self.decodedText += '"'
+
+                self.decodedText += f'\\n"\n'
+
+                prevText = False
+                prevNewline = True
+                prevCmd = False
+            elif char == 0x04 or char == 0x0C: # Box Breaks add automatic newlines
+                if prevText:
+                    self.decodedText += '"'
+
+                cmd, decoder = CMDMAP[char]
+                self.decodedText += f'\n{decoder(cmd)}\n'
+
+                prevText = False
+                prevNewline = True
+                prevCmd = False
+            elif char == 0x02: # End command, do nothing
+                if prevText:
+                    self.decodedText += '"'
+            else: # Control Codes (see message_data_fmt_staff.h)
+                if prevText:
+                    self.decodedText += '" '
+                elif prevCmd:
+                    self.decodedText += ' '
+
+                cmd, decoder = CMDMAP[char]
+                self.decodedText += decoder(cmd)
+
+                prevText = False
+                prevNewline = False
+                prevCmd = True
+
+def parseTable(start):
+    table = {}
+    with open("baserom/code","rb") as f:
+        f.seek(start)
+        buf = f.read(8)
+        textId, typePos, segment = struct.unpack(">HBxI", buf)
+        while segment != 0:
```

Comment:
Reminder to self: Actually check if textId is` 0xFFFF` instead of segment is 0 like `Message_FindMessageCredits`